### PR TITLE
Updated IncrementStrategyFinder to support version bumping from tags

### PR DESCRIPTION
**Pull request by MikeLarah**
Updated IncrementStrategyFinder to support version bumping from tagged commits.

Updated tests and docs.

#971

### CODE REVIEW

File: src/GitVersionCore/Configuration/Config.cs [hunk 1]
```diff
@@ -55,6 +55,19 @@ public string NextVersion
...
+        [YamlMember(Alias = "minor-version-bump-tag")]
+        public string MinorVersionBumpTag { get; set; }
+
+        [YamlMember(Alias = "patch-version-bump-tag")]
+        public string PatchVersionBumpTag { get; set; }
+
+        [YamlMember(Alias = "no-bump-tag")]
+        public string NoBumpTag { get; set; }
+
+
```
Comment (pascalberger) [inline src/GitVersionCore/Configuration/Config.cs]: Remove double linebreak

**Comment by pascalberger** [comment]
Thanks for your contribution! LGTM

@asbjornu @gep13 @JakeGinnivan OK for you too merge this?

**Comment by JakeGinnivan** [comment]
Would you be able to walk me through a typical workflow on how you would use this feature @MikeLarah?

**Comment by JakeGinnivan** [comment]
@MikeLarah am keen to hear more about the requirements for this change?

**Comment by MikeLarah** [comment]
@JakeGinnivan Sorry for the delayed response. Our intention was to use this in combination with a process in the CI build that determined whether the public API has been changed.
e.g. 1) developer commits and pushes up changes to a feature branch, 2) CI build does comparison between feature branch binaries and latest master binaries to determine if API has changed, 3) we tag the commit with relevant major/minor increment tag if necessary, 4) future builds are automatically version incremented using GitVersion (and will be correctly incremented when the feature is merged to develop branch)

**Comment by JakeGinnivan** [comment]
Hey Mike,

I see a few issues with this approach

1. More config :(
2. `git tag` will have a heap of `+semver-minor` tags, which is noisy

Alternatives:

When you merge pull requests, you can alter the merge message in every tool I know. At that point simply append `+semver: minor` to the merge PR commit message.

Then you can use the new Mainline development mode in v4 to aggregate the versions from each merged PR. I think your process is working around the fact mainline was missing and I think it should be able to solve your problem. Let me know how you go, going to close this out for now (but we can keep the convo open)

**Pull request closed without merging**